File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    answer = 0
    queue = deque([(p, idx) for idx, p in enumerate(priorities)])
    
    while queue:
        current = queue.popleft()
        
        if any(current[0] < q[0] for q in queue):
            queue.append(current)
        else:
            answer += 1
            if current[1] == location:
                return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        
        Queue<int[]> queue = new LinkedList<>();
        for (int i=0; i<priorities.length; i++) {
            queue.offer(new int[]{i, priorities[i]});
        }
        
        int count = 0;
        while (!queue.isEmpty()) {
            int[] current = queue.poll();
            boolean hasHigher = false;
            
            for (int[] q: queue) {
                if (q[1] > current[1]) {
                    hasHigher = true;
                    break;
                }
            }
            
            if (hasHigher) {
                queue.offer(current);
            }
            else {
                count ++;
                if (current[0] == location) {
                    return count;
                }
            }
        }
        return -1;
    }
}